### PR TITLE
user details screen

### PR DESCRIPTION
**Pull request by Ruthy4**

### CODE REVIEW

File: app/src/main/java/com/example/randomusergenerator/user/UserScreen.kt [hunk 1]
```diff
@@ -13,25 +13,28 @@ import androidx.compose.runtime.*
...
 import com.example.randomusergenerator.user.viewmodel.UserViewModel
 
 @Composable
 fun UserScreen() {
-    val userViewModel: UserViewModel = viewModel()
+    val userViewModel: UserViewModel = hiltViewModel()
     val userViewState by userViewModel.uiState.collectAsState()
 
-    LaunchedEffect(key1 = true) {
+    LaunchedEffect(key1 = false) {
```
Comment (princessdharmy) [inline app/src/main/java/com/example/randomusergenerator/user/UserScreen.kt:28]: Why change to `false` if I may ask? 🤔 
And why aren't you using a string as a key identifier?

File: app/src/main/java/com/example/randomusergenerator/view/userdetails/UserDetails.kt [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        horizontalAlignment = Alignment.CenterHorizontally
+    ) {
+
+        Image(
+            modifier = Modifier
+                .align(Alignment.Start)
+                .size(space_x5)
+                .clickable { userViewModel.navigateBack() },
+            painter = painterResource(id = R.drawable.back_arrow),
+            contentDescription = "navigate back"
```
Comment (princessdharmy) [inline app/src/main/java/com/example/randomusergenerator/view/userdetails/UserDetails.kt:54]: Extract string.

File: app/src/main/java/com/example/randomusergenerator/user/UserScreen.kt (hunk 1 above)
Comment (Ruthy4) [reply app/src/main/java/com/example/randomusergenerator/user/UserScreen.kt:28]: I was checking for why it always calls the getUser function whenever the user comes to first screen. Still same behaviour when the navigateUP is used to navigate back

Comment (princessdharmy) [reply app/src/main/java/com/example/randomusergenerator/user/UserScreen.kt]: I will move this discussion to slack.

Comment (Ruthy4) [reply app/src/main/java/com/example/randomusergenerator/user/UserScreen.kt]: okay

File: app/src/test/java/com/example/randomusergenerator/user/viewmodel/UserViewModelTest.kt [hunk 3]
```diff
@@ -82,4 +84,16 @@ class UserViewModelTest {
         assertTrue(actualResult.isLoading)
         assertEquals(true, actualResult.isLoading)
     }
+
+    @Test
+    fun `when navigateToUserDetails is called, then verify correct function is called`() = runTest {
```
Comment (princessdharmy) [inline app/src/test/java/com/example/randomusergenerator/user/viewmodel/UserViewModelTest.kt:89]: Kindly remove the comma from the function name. I overlooked it previously but had to research it.

Comment (Ruthy4) [reply app/src/test/java/com/example/randomusergenerator/user/viewmodel/UserViewModelTest.kt:89]: okay

**Review by princessdharmy — APPROVED** [review]

**Pull request merged**